src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,245 @@
+package parser;
+
+
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import chatterboxexceptions.ChatterboxExceptions;
+
+/**
+ * Parser class has various methods to parse user input
+ */
+public class Parser {
+    /**
+     * Enum contains list of valid commands
+     */
+    public enum ValidCommand {
+        BYE, LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, INVALID, FIND;
+    }
+
+
+    private static final DateTimeFormatter DASHFORMATTER = DateTimeFormatter.ofPattern("dd-MM-yyyy HHmm");
+    private static final DateTimeFormatter SLASHFORMATTER = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+
+    private static final DateTimeFormatter DASHONLYDATE = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+    private static final DateTimeFormatter SLASHONLYDATE = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+    private static final DateTimeFormatter PRINTDATEFORMATTER = DateTimeFormatter.ofPattern("MMM dd yyyy, HH:mm");
+    private static final DateTimeFormatter[] DATE_TIME_FORMATTERS =
+            new DateTimeFormatter[] {DASHFORMATTER, SLASHFORMATTER, PRINTDATEFORMATTER};
+    private static final DateTimeFormatter[] DATE_ONLY_FORMATTERS =
+            new DateTimeFormatter[] {DASHONLYDATE, SLASHONLYDATE};
```

Comment:
A bit inconsistent, it is better to use <code>DASH_ONLY_DATE</code> instead of <code>DASHONLYCODE</code> for consistency. This also applies for another static variable